src/Voron/Impl/Paging/AbstractPager.cs

```diff
@@ -261,8 +260,64 @@ public static bool IsKeySizeValid(int keySize)
             return true;
         }
 
-        public abstract void MaybePrefetchMemory(List<long> pagesToPrefetch);
+        protected List<Win32MemoryMapNativeMethods.WIN32_MEMORY_RANGE_ENTRY> SortedPagesToList(List<TreePage> sortedPages)
+        {
+            var rangesList = new List<Win32MemoryMapNativeMethods.WIN32_MEMORY_RANGE_ENTRY>();
+
+            long lastPage = -1;
+            const int numberOfPagesInBatch = 8;
+            var sizeInPages = numberOfPagesInBatch; // OS uses 32K when you touch a page, let us reuse this
+            foreach (var page in sortedPages)
+            {
+                if (lastPage == -1)
+                {
+                    lastPage = page.PageNumber;
+                }
+
+                var numberOfPagesInLastPage = page.IsOverflow == false
+                    ? 1
+                    : this.GetNumberOfOverflowPages(page.OverflowSize);
+
+                var endPage = page.PageNumber + numberOfPagesInLastPage - 1;
+
+                if (endPage <= lastPage + sizeInPages)
+                    continue; // already within the allocation granularity we have
+
+                if (page.PageNumber <= lastPage + sizeInPages + numberOfPagesInBatch)
+                {
+                    while (endPage > lastPage + sizeInPages)
+                    {
+                        sizeInPages += numberOfPagesInBatch;
+                    }
+
+                    continue;
+                }
+
+                rangesList.Add(new Win32MemoryMapNativeMethods.WIN32_MEMORY_RANGE_ENTRY
+                {
+                   VirtualAddress = AcquirePagePointer(null, lastPage),
+                    NumberOfBytes = (IntPtr)(sizeInPages * PageSize)
```

Comment:
where are you accounting for the granularity? If I'm loading 10 pages, I'm actually loading 16.